src/Hashids.php

```diff
@@ -80,7 +80,7 @@ public function __construct(
     /**
      * Encode parameters to generate a hash.
      *
-     * @param string|array<int, string> $numbers
+     * @param int|string|array<int, int|string> $numbers
```

Comment:
I'm guessing `<int, int|string>` should be replaced with `<int|string>`?

Reply:
no. i think `array<int, int|string>` is correct.

Reply:
Hmm, why isn't the syntax `<int, string, int|string>`?

Reply:
think about it in the following order:

- $numbers only accepts type int: `@param int $numbers`
- it also accepts string types: `@param int|string $numbers`
- it also accepts arrays whose values are of type int: `@param int|string|array<int, int> $numbers`
- it also accepts arrays whose values are strings: `@param int|string|array<int, int|string> $numbers`

the first int type in the array represents the key type. that is `array<array-key, array-value>`.

```php
$foo = [1, 2, 3];
var_export($foo);
// array (0 => 1, 1 => 2, 2 => 3)

$bar = ["1", "2", "3"];
var_export($bar);
// array ( 0 => '1', 1 => '2', 2 => '3')
```

looking at the readme, i don't see anything like using strings as array keys. anything you don't expect should be protected by defining an explicit type.

are you thinking of using it in the following ways?

```php
$hashids->encode(["foo" => 1, "bar" => 2]);
```

if not, you should explicitly define the array keys to be of type int.

Reply:
Thanks for a thorough explanation 🙌